src/test/java/com/slack/kaldb/writer/kafka/KaldbKafkaConsumerTest.java

```diff
@@ -101,11 +104,20 @@ public void kafkaConsumerStartupShutdown() throws Exception {
     @Before
     public void setUp() throws Exception {
       kafkaServer = new TestKafkaServer();
-      testConsumer = new TestKaldbKafkaConsumer(kafkaServer.getBroker().getBrokerList().get());
+      testConsumer =
+          new TestKaldbKafkaConsumer(
+              kafkaServer.getBroker().getBrokerList().get(), "timeout_tests");
     }
 
     @After
-    public void tearDown() {}
+    public void tearDown() throws ExecutionException, InterruptedException, TimeoutException {
+      // KafkaConsumer.DEFAULT_CLOSE_TIMEOUT_MS is 30 * 1000, so wait that plus a little extra
+      with()
+          .pollInterval(1, TimeUnit.SECONDS)
+          .await()
+          .atMost(45, TimeUnit.SECONDS)
+          .until(() -> testConsumer.isShutdown());
```

Comment:
Could the threads be hanging since we forgot to shutdown the consumer properly?

Is there a reason the tearDown function here is different than the one on line 71? Can we use the same method here? Also, do we need to add this logic for the above test also?

Reply:
In the timeout test we intentionally close the `kafkaServer` before the consumer to replicate a timeout. We do request to close the consumer immediately after the server (`testConsumer.triggerShutdown()`), but this is not a synchronous call. 

This change ensures we wait until the consumer has completely closed before continuing onto the next test. The other test does not require this, as we close in expected order and it closes cleanly without queuing a reconnection.